packages/shared/profiles/sagas.ts

```diff
@@ -150,19 +155,16 @@ export function* handleFetchProfile(action: ProfileRequestAction): any {
     const shouldReadProfileFromLocalStorage = yield select(isCurrentUserId, userId)
     const shouldFetchViaComms = commsContext && !shouldReadProfileFromLocalStorage
     const shouldLoadFromCatalyst = true
-    const shouldFallbackToRandomProfile = true
 
     const profile: Avatar =
       // first fetch avatar through comms
       (shouldFetchViaComms && (yield call(requestProfileToPeers, commsContext, userId, version))) ||
       // and then via catalyst
       (shouldLoadFromCatalyst && (yield call(getRemoteProfile, userId, version))) ||
       // then for my profile, try localStorage
-      (shouldReadProfileFromLocalStorage && (yield call(readProfileFromLocalStorage))) ||
-      // lastly, come up with a random profile
-      (shouldFallbackToRandomProfile && (yield call(generateRandomUserProfile, userId)))
```

Comment:
Where does this fallback take place?

Reply:
[Here](https://github.com/decentraland/kernel/pull/478/files#diff-0119c0e28687b0c42257febe4d02be417a56f511d1582182c47f25076871bb3dR285).

Reply:
can this be moved back here? otherwise the logic to resolve profiles is hard to follow deep inside the stack trace. it is not clear enough that `getRemoteProfile` generates a profile if it is not found